pkgs/stdenv/generic/setup.sh

```diff
@@ -334,36 +334,44 @@ appendToVar() {
     fi
 }
 
-# Accumulate into `flagsArray` the flags from the named variables.
+# Accumulate flags from the named variables $2+ into the array $1.
 #
-# If __structuredAttrs, the variables are all treated as arrays
-# and simply concatenated onto `flagsArray`.
-#
-# If not __structuredAttrs, then:
-#   * Each variable is treated as a string, and split on whitespace;
-#   * except variables whose names end in "Array", which are treated
-#     as arrays.
-_accumFlagsArray() {
-    local name
-    if [ -n "$__structuredAttrs" ]; then
-        for name in "$@"; do
-            local -n nameref="$name"
-            flagsArray+=( ${nameref+"${nameref[@]}"} )
-        done
-    else
-        for name in "$@"; do
+# Arrays are simply concatenated, strings are split on whitespace.
+concatTo() {
+    local -n targetref="$1"; shift
+    local name type
+    for name in "$@"; do
+        if type=$(declare -p "$name" 2> /dev/null); then
             local -n nameref="$name"
-            case "$name" in
-                *Array)
-                    # shellcheck disable=SC2206
-                    flagsArray+=( ${nameref+"${nameref[@]}"} ) ;;
+            case "${type#* }" in
+                -A*|-a*)
+                    targetref+=( "${nameref[@]}" ) ;;
                 *)
                     # shellcheck disable=SC2206
-                    flagsArray+=( ${nameref-} ) ;;
+                    targetref+=( ${nameref-} ) ;;
             esac
-        done
-    fi
+        fi
+    done
+}
 
+# Separates a list of things by $1 separator.
+# For arrays each item is separated, for strings whitespace is replaced.
+# To be used for partial support of structuredAttrs.
+separateBy() {
```

Comment:
I'm not sure I understand what's happening here.
I only see one usage example, with `mesonInstallTags`.
Looking at the nixpkgs manual, the hook expects the user to specify a list of strings, and promises to concatenate the strings, joined by commas.

- The way I read the comment it does the opposite, i.e. it splits, not concats
- I'm not sure that either concatenation or the splitting are what `IFS=, echo "${nameref[*]}"` does, at least it's not what I see when I try it in repl?
- If this were to do the splitting, I don't see why you'd ever want to destructure an array in a hook?
- Also not obvious from the description that the result goes into stdout

Reply:
`concatStringsSep` would be familiar from nix, `join` might be familiar to python users

Reply:
I'm still concerned about the behaviour with the structured attrs on